1-JavaScript/Week3/homework/ex4-shoppingCart.js

```diff
@@ -21,8 +21,19 @@ you have more than 3 items in your shopping cart the first item gets taken out.
 -----------------------------------------------------------------------------*/
 const shoppingCart = ['bananas', 'milk'];
 
-function addToShoppingCart(/* parameters go here */) {
-  // TODO complete this function
+function addToShoppingCart(grocery) {
+  if (shoppingCart.length > 2) {
+    shoppingCart.shift();
+    shoppingCart.push(grocery);
+  } else {
+    shoppingCart.push(grocery);
+  }
+  let i = 0;
+  for (i = 0; i < shoppingCart.length; i++) {
```

Comment:
Consider the scope `i` will be available and used in.  Is there an advantage to declaring it before it is used in the for loop?

Reply:
You are right, there is no advantage  to declare it before loop, just some old habits from Java :) 
I changed it. as you said.